src/torchcodec/_core/SingleStreamDecoder.cpp

```diff
@@ -446,13 +441,12 @@ void SingleStreamDecoder::addVideoStream(
   auto& streamMetadata =
       containerMetadata_.allStreamMetadata[activeStreamIndex_];
 
-  if (seekMode_ == SeekMode::approximate &&
-      !streamMetadata.averageFpsFromHeader.has_value()) {
-    throw std::runtime_error(
-        "Seek mode is approximate, but stream " +
-        std::to_string(activeStreamIndex_) +
-        " does not have an average fps in its metadata.");
-  }
+  TORCH_CHECK(
+      !(seekMode_ == SeekMode::approximate &&
+        !streamMetadata.averageFps.has_value()),
+      "Seek mode is approximate, but stream ",
```

Comment:
I find this boolean expression much harder to reason about than the previous one. It's wordier, but let's instead do:
```c++
if (seekMode_ == SeekMode::approximate) {
  TORCH_CHECK(
    streamMetadata.averageFpsFromHeader.has_value(),
    "Same message as before");
}
```
I prefer that, as it reads closer to what we would say out loud: if the seek mode is approximate, assert that it has a value for fps.